### PR TITLE
Split pkg/idtools

### PR DESCRIPTION
**Pull request by vieux**
See #32989

Split `"github.com/docker/docker/pkg/idtools` between `github.com/docker/docker/pkg/fsutils1 and `github.com/docker/docker/pkg/system`

### CODE REVIEW

**Comment by dnephin** [comment]
This is going to conflict heavily with #33362. I hope we can merge that first. It's a more involved change. Sorry I should have put that into the tracking issue.

**Comment by vieux** [comment]
@dnephin ok no worries

**Comment by thaJeztah** [comment]
ping @vieux - https://github.com/moby/moby/pull/33362 was merged, so this needs an, erm, rebase or rewrite

**Comment by vieux** [comment]
@dnephin rebased.

File: pkg/system/utils_unix.go [hunk 1]
```diff
@@ -9,7 +9,8 @@ import (
 	"strings"
 )
 
-func resolveBinary(binname string) (string, error) {
+// ResolveBinary resolves a binary path (synlinks and such).
```
Comment (thaJeztah) [inline pkg/system/utils_unix.go]: s/synlinks/symlinks/

**Review by thaJeztah — APPROVED** [review]
LGTM

spotted one typo

File: pkg/system/utils_unix.go [hunk 2]
```diff
@@ -26,7 +27,8 @@ func resolveBinary(binname string) (string, error) {
 	return "", fmt.Errorf("Binary %q does not resolve to a binary of that name in $PATH (%q)", binname, resolvedPath)
 }
 
-func execCmd(cmd, args string) ([]byte, error) {
+// ExecCmd calls exec.Command and returns the CombinedOutput.
+func ExecCmd(cmd, args string) ([]byte, error) {
```
Comment (dnephin) [inline pkg/system/utils_unix.go]: I don't think we should export this two line helper.

I believe with the suggested split up  this won't need to be exported anymore, as the only consumers will be in `pkg/system/user`

File: pkg/fsutils/usergroupadd_linux.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package idtools
+package fsutils
```
Comment (dnephin) [inline pkg/fsutils/usergroupadd_linux.go]: This is the part of the package that I thought would move to `pkg/system`. It doesn't have much to do with files.

This file could move to `pkg/system/user/usergroupadd_linux.go`

File: pkg/fsutils/usergroupadd_unsupported.go [hunk 4]
```diff
@@ -1,6 +1,6 @@
 // +build !linux
 
-package idtools
+package fsutils
```
Comment (dnephin) [inline pkg/fsutils/usergroupadd_unsupported.go]: This file could move to `pkg/system/user/....go`

File: pkg/fsutils/idtools_unix.go [hunk 5]
```diff
@@ -1,6 +1,6 @@
 // +build !windows
 
-package idtools
+package fsutils
```
Comment (dnephin) [inline pkg/fsutils/idtools_unix.go]: This file needs to be split up.

`LookupX()` would go to `pkg/system/user/usergroupadd_linux.go`

The file stiff can stay in fsutil

File: pkg/fsutils/idtools.go [hunk 6]
```diff
@@ -1,4 +1,4 @@
-package idtools
+package fsutils
```
Comment (dnephin) [inline pkg/fsutils/idtools.go]: I think some of this file needs to be split up.

All the Mkdir... would be fsutils, but the ID stuff probably makes more sense in system/user/

The file stiff can stay in fsutil

File: pkg/system/utils_unix.go [hunk 7]
```diff
@@ -9,7 +9,8 @@ import (
 	"strings"
 )
 
-func resolveBinary(binname string) (string, error) {
+// ResolveBinary resolves a binary path (synlinks and such).
+func ResolveBinary(binname string) (string, error) {
```
Comment (dnephin) [inline pkg/system/utils_unix.go]: I believe with the suggested split up  this won't need to be exported anymore, as the only consumers will be in `pkg/system/user`

**Review by dnephin — CHANGES_REQUESTED** [review]

File: pkg/system/utils_unix.go (hunk 2 above)
Comment (vieux) [inline pkg/system/utils_unix.go]: @dnephin so you would move the whole `utils_unix.go` to `pkg/system/user` ?

Comment (dnephin) [inline pkg/system/utils_unix.go]: The file is only these 2 functions, right? so yes I think it would all move to `pkg/system/user`.

**Comment by thaJeztah** [comment]
ping @vieux this needs a rebase

**Comment by vieux** [comment]
@thaJeztah yup, @dnephin will carry the PR, assigning to him.

**Comment by thaJeztah** [comment]
ping @dnephin ^^

**Comment by dnephin** [comment]
This seems fairly low priority given the new path for the engine. I'm going to close this PR for now. It's still on my list.

**Pull request closed without merging**